hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersTranslator.java

```diff
@@ -148,21 +148,21 @@ public ClassicTransfersCall callFrom(@NonNull final HtsCallAttempt attempt) {
 
     private @Nullable TransactionBody nominalBodyFor(@NonNull final HtsCallAttempt attempt) {
         if (attempt.isSelector(CRYPTO_TRANSFER)) {
-            return decoder.decodeCryptoTransfer(attempt.inputBytes(), attempt.addressIdConverter());
+            return decoder.decodeCryptoTransfer(attempt.inputBytes(), attempt);
```

Comment:
Out of curiosity, why are both the `entityIdFactory` _and_ the `addressIdConverter` necessary inside of `decodeCryptoTransfer(...)`? From a purely conceptual standpoint, it seems like `addressIdConverter` alone could handle the translation (including shard/realm). Maybe its job is more specific than this, though?

Reply:
The `addressIdConverter` encapsulates additional logic having to do with whether the id we are creating is being credited or debit value.  Nevertheless, this is a good point as the `addressIdConverter` has access to the `entityIdFactory`.

I think at the end of the day, I converted everything to `attempt` classes because this gives the inner calls access to everything we know about the call attempt rather than just the classes for doing id conversions.